indra/tests/test_belief_sklearn.py

```diff
@@ -0,0 +1,345 @@
+import random
+import pickle
+import numpy as np
+from copy import copy
+from collections import defaultdict
+from os.path import join, abspath, dirname
+from nose.tools import raises
+from sklearn.linear_model import LogisticRegression
+from indra.sources import signor
+from indra.belief import BeliefEngine
+from indra.tools import assemble_corpus as ac
+from indra.statements import Evidence
+from indra.belief.skl import CountsScorer
+
+
+# A set of test statements derived from SIGNOR only
+# (these include many different stmt types)
+test_stmt_path = join(dirname(abspath(__file__)),
+                      'belief_sklearn_test_stmts.pkl')
+
+
+# An alternative set of test statements derived from the curated stmt dataset
+# (these include supports/supported_by)
+test_stmt_cur_path = join(dirname(abspath(__file__)),
+                          'belief_sklearn_test_stmts_cur.pkl')
+
+
+# A statement dataframe sample
+test_df_path = join(dirname(abspath(__file__)),
+                    'belief_sklearn_test_df.pkl')
+
+
+with open(test_stmt_path, 'rb') as f:
+    test_stmts, y_arr_stmts = pickle.load(f)
+
+
+with open(test_stmt_cur_path, 'rb') as f:
+    test_stmts_cur, y_arr_stmts_cur = pickle.load(f)
+
+
+with open(test_df_path, 'rb') as f:
+    test_df, y_arr_df = pickle.load(f)
+
+
+# A set of statements derived from Signor used for testing purposes.
+def _dump_test_data(filename, num_per_type=10):
+    """Get corpus of statements for testing that has a range of stmt types."""
+    sp = signor.process_from_web()
+    # Group statements by type
+    stmts_by_type = defaultdict(list)
+    for stmt in sp.statements:
+        stmts_by_type[stmt.__class__].append(stmt)
+    # Sample statements of each type (without replacement)
+    stmt_sample = []
+    for stmt_type, stmt_list in stmts_by_type.items():
+        if len(stmt_list) <= num_per_type:
+            stmt_sample.extend(stmt_list)
+        else:
+            stmt_sample.extend(random.sample(stmt_list, num_per_type))
+    # Make a random binary class vector for the stmt list
+    y_arr = [random.choice((0, 1)) for s in stmt_sample]
+    with open(test_stmt_path, 'wb') as f:
+        pickle.dump((stmt_sample, y_arr), f)
+    return stmt_sample
+
+
+def test_counts_wrapper():
+    """Instantiate counts wrapper and make stmt matrix"""
+    lr = LogisticRegression()
+    source_list = ['reach', 'sparser']
+    cw = CountsScorer(lr, source_list)
```

Comment:
perhaps make `lr` optional in CountsScorer and internally instantiate a LogisticRegression by default

Reply:
Since the best approach to modeling/fitting statement data is still in research mode, I think it's probably better to keep this explicit at least for now